src/z3c/recipe/sphinxdoc/__init__.py

```diff
@@ -168,9 +171,11 @@ def install(self):
     update = install
 
 
-def main(projects):
+def main(projects, argv=None, exit_on_error=False):
```

Comment:
> and I think this script should exit with non-zero if documentation generation fails.

I wouldn't argue with that, *if* it tried to generate all the docs before exiting. So something like:
```python
code = 0
for project in projects:
   code |= sphinx.build()
sys.exit(code)
```

(Except I find it obnoxious when functions call `sys.exit` because it makes them annoying to unittest. I tend to reserve that for the `if __name__ == '__main__':` block. Which doesn't quite work here.)